src/algokit_utils/applications/app_deployer.py

```diff
@@ -579,58 +581,87 @@ def _handle_schema_break(
         existing_app: ApplicationMetaData,
         approval_program: bytes,
         clear_program: bytes,
     ) -> AppDeployResult:
+        suppress_log = deployment.send_params.get("suppress_log") or False if deployment.send_params else False
+
         if deployment.on_update in (OnUpdate.Fail, "fail") or deployment.on_update is None:
             raise ValueError(
-                "Update detected and onUpdate=Fail, stopping deployment. " "Try a different onUpdate value to not fail."
+                "Update detected and on_update=Fail, stopping deployment. "
+                "Try a different on_update value to not fail."
             )
 
         if deployment.on_update in (OnUpdate.AppendApp, "append"):
+            config.logger.info(
+                "on_update=AppendApp, will attempt to create a new app", extra={"suppress_log": suppress_log}
+            )
             return self._create_app(deployment, approval_program, clear_program)
 
         if deployment.on_update in (OnUpdate.UpdateApp, "update"):
             if existing_app.updatable:
-                return self._update_app(deployment, existing_app, approval_program, clear_program)
+                config.logger.info(
+                    "App is updatable and on_update=UpdateApp, updating app...",
+                    extra={"suppress_log": suppress_log},
+                )
             else:
-                raise ValueError(
-                    f"App is {'not' if not existing_app.updatable else ''} updatable and onUpdate=UpdateApp, "
-                    "cannot update app"
+                config.logger.info(
+                    "App is not updatable but on_update=UpdateApp, will attempt to update app, "
+                    "update will most likely fail",
+                    extra={"suppress_log": suppress_log},
                 )
+            return self._update_app(deployment, existing_app, approval_program, clear_program)
 
         if deployment.on_update in (OnUpdate.ReplaceApp, "replace"):
             if existing_app.deletable:
-                return self._replace_app(deployment, existing_app, approval_program, clear_program)
+                config.logger.info(
+                    "App is deletable and on_update=ReplaceApp, will attempt to create new app and delete old app",
+                    extra={"suppress_log": suppress_log},
+                )
             else:
-                raise ValueError(
-                    f"App is {'not' if not existing_app.deletable else ''} deletable and onUpdate=ReplaceApp, "
-                    "cannot delete and recreate app"
+                config.logger.info(
+                    "App is not deletable but on_update=ReplaceApp, will attempt to create a new app and "
+                    "delete the old app, delete will most likely fail",
+                    extra={"suppress_log": suppress_log},
                 )
+            return self._replace_app(deployment, existing_app, approval_program, clear_program)
```

Comment:
The logic change here is a big confusing and doesn't call out why this change was needed.

I think a better approach would be to patch the existing_app after it's fetched to handle the scenario. We could then wrap a nice comment with the why.

This logic could then stay the same.

The logic would look something like this:
```python
  # V2 of utils had a behaviour where...
  if existing_app and existing_app.deletable is None:
      existing_app = dataclasses.replace(existing_app, deletable=True)
  
  if existing_app and existing_app.updatable is None:
      existing_app = dataclasses.replace(existing_app, updatable=True)

```

Reply:
Also just to confirm the v2 behaviour. The `deletable`/`updatable` flags behaved as below:
- `True` -> arc2 note would contain  `true`
- `False` -> arc2 note would contain  `false`
- `None` -> arc2 note would contain  `null`

If `deletable`/`updatable` is explicitly False, then we'd want to respect that setting.